### PR TITLE
enhance(main/ncurses): drop `termite`, add `foot` terminfo

### PR DESCRIPTION
**Pull request by TomJo2000**
There's 3 main parts to this PR.
- 1.) Drop `termite`'s terminfo.
The `termite` project has been [archived for close to 3 and a half years](https://github.com/thestinger/termite).
We can probably stop shipping a terminfo file for it.
  - 1.1) We *do* probably wanna start shipping terminfo files for `foot`, since we have it packaged and it is in very active development.
- 2.) Make "version realignment" for `alacritty`, `foot` and `kitty` easier.
Currently keeping version and checksum for these synced between their build scripts and `ncurses` is a pain in the ass.
Since what we need from them is the `TERMUX_PKG_{VERSION,SRCURL,SHA256}` values, let's just grab them from the respective build script.
This way they can also be "safely" auto-updated, without creating additional work to "realign" the values in the `ncurses` build.
- 3.) General Shellcheck warning cleanup.
Quoting, more descriptive error messages, etc.

I'm gonna leave enabling auto-updates on `alacritty` and `kitty` to another PR.
This one is already dealing with enough moving parts as is.

### CODE REVIEW

File: packages/ncurses/build.sh
Comment (twaik) [inline packages/ncurses/build.sh:128]: Are you listing all possible terminals here? Because there are kermit and xfce4-terminal unlisted.

Comment (twaik) [reply packages/ncurses/build.sh:128]: And mate-terminal too. I am not sure these terminals need their own terminfo.

**Review by twaik — APPROVED** [review]

File: packages/ncurses/build.sh
Comment (TomJo2000) [reply packages/ncurses/build.sh:128]: I didn't add any here compared to what we ship currently.
But all three of those terminals are VTE based, so they shouldn't need a bespoke terminfo.
The `gnome` one should apply.

**Review by truboxl — APPROVED** [review]

**Comment by TomJo2000** [comment]
I'd like to merge this in ~6 hours if there's no other comments or concerns.

**Comment by TomJo2000** [comment]
@fornwall looks like you started working on the same thing ~5 months ago and never finished it.
https://github.com/termux/termux-packages/blob/ncurses-kitty-alacritty/packages/ncurses/build.sh

We can delete the `ncurses-kitty-alacritty` branch after this PR is merged.

**Comment by TomJo2000** [comment]
I'm gonna merge this now so it's ready in time for the next autoupdate window in ~30 minutes.
That way `alacritty` can get autoupdated pretty much immediately after #21961 is merged.

**Pull request merged**
